src/module_wrap.h

```diff
@@ -187,6 +192,26 @@ class ModuleWrap : public BaseObject {
       const v8::FunctionCallbackInfo<v8::Value>& args);
   static void CreateCachedData(const v8::FunctionCallbackInfo<v8::Value>& args);
 
+  static ModuleWrap* GetFromModule(node::Environment*, v8::Local<v8::Module>);
+
+  v8::Global<v8::Module> module_;
+  contextify::ContextifyContext* contextify_context_ = nullptr;
+  bool synthetic_ = false;
+  bool linked_ = false;
+  int module_hash_;
+
+  friend class ModuleInstantiationContext;
+};
+
+class ModuleInstantiationContext {
+  using ResolveCache = std::unordered_map<ModuleCacheKey,
+                                          v8::Global<v8::Object>,
+                                          ModuleCacheKey::Hash>;
+  // The key (a pointer to ModuleWrap) can be sorted, so we use a map for
+  // efficiency.
```

Comment:
This comment is a bit confusing, we don't actually seem to be making use of the sorting – if you're talking about the relative performance of `std::unordered_map` and `std::map` in general, that depends on quite a few factors and it's not as simple as saying "`std::map` is better for sortable keys and `std::unordered_map` for others". I'd generally use `std::unordered_map` by default _unless_ there is a need specifically for sorting semantics or I have very concrete reason to believe that `std::map` would perform better (e.g. for a small number of keys)